src/channel.rs

```diff
@@ -46,26 +49,29 @@ impl Channel {
             .is_some_and(|alias| matches!(alias, ChannelAlias::Nightly(None)))
     }
 
+    /// This functions compares the Channel &self, with a newer channel [newer]
+    /// and returns the list of [Components] that need to be updated.
     pub fn components_to_update(&self, newer: &Self) -> Vec<Component> {
-        let new: HashSet<&Component> = HashSet::from_iter(newer.components.iter());
+        let new_channel: HashSet<&Component> = HashSet::from_iter(newer.components.iter());
         let current: HashSet<&Component> = HashSet::from_iter(self.components.iter());
```

Comment:
```suggestion
        let new_channel = HashSet::from_iter(newer.components.iter());
        let current = HashSet::from_iter(self.components.iter());
```

The types here can be inferred, so I would prefer to elide them. As a general rule, explicit types are worth adding when it isn't clear what the type _should_ be inferred as, or when the compiler can't infer them - the rest of time there's little reason to annotate them.

Reply:
>  or when the compiler can't infer them 

I think this is case in this particular snippet. I committed the change to showcase the error, [here](https://github.com/0xMiden/midenup/actions/runs/16194237056/job/45716741067#step:6:133). (Sidenote: I'll remove this commit before merging).

I think this is the same problem as the one described in this [StackOverflow question](https://stackoverflow.com/q/62949404). This seems to be a problem with `from_iter`, here's a little recreation in the playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=3aa55152947fe266af1dc647107b2a4a

Interestingly enough, only one explicit type hint is needed to satisfy the compiler. The following does compile: 
```
    pub fn components_to_update(&self, newer: &Self) -> Vec<Component> {
        let new_channel: HashSet<&Component> = HashSet::from_iter(newer.components.iter());
        let current = HashSet::from_iter(self.components.iter());
```
Although, I decided against it originally because it seemed a bit inconsistent. 


With that in mind, I pushed this commit: https://github.com/0xMiden/midenup/pull/27/commits/47af5594ac0379107e337c4b9106a42e3f0b1a88 which removed the inner type of the `HashSet`.

Let me know which version you prefer!

Reply:
Ah interesting! I suspect it's because of how we use `new_channel` vs `current`; if you try to interact with an object with an inferred type in a way that also requires inference, the compiler complains and wants you to specify a type. I think in this case, `current` avoids the need for the type annotation because we are passing it as an argument to a `HashSet` method that requires a specific concrete type, but only if the type of `new_channel` is known, so it complains about one but not the other.

I think it's fine to minimize the type annotations, even if it looks a bit weird to annotate one but not the other, it's essentially just to satisfy the compiler in this case.

Reply:
> Ah interesting! I suspect it's because of how we use `new_channel` vs `current`; if you try to interact with an object with an inferred type in a way that also requires inference, the compiler complains and wants you to specify a type. I think in this case, `current` avoids the need for the type annotation because we are passing it as an argument to a `HashSet` method that requires a specific concrete type, but only if the type of `new_channel` is known, so it complains about one but not the other.
> 
> I think it's fine to minimize the type annotations, even if it looks a bit weird to annotate one but not the other, it's essentially just to satisfy the compiler in this case.

Noted! Changed it here: https://github.com/0xMiden/midenup/commit/75107e41c834ec86428a3900e0c51452b83a7646. With this now in place, I think this is ready to get merged 😊